src/panels/config/cloud/account/cloud-google-pref.ts

```diff
@@ -148,24 +176,25 @@ export class CloudGooglePref extends LitElement {
     `;
   }
 
-  private async _syncEntitiesCalled(ev: CustomEvent) {
-    if (!ev.detail.success && ev.detail.response.status_code === 404) {
-      this._syncFailed();
+  private async _handleSync() {
+    this._syncing = true;
+    try {
+      await syncCloudGoogleEntities(this.hass!);
+    } catch (err: any) {
+      showAlertDialog(this, {
+        title: this.hass.localize(
+          "ui.panel.config.cloud.account.google.not_configured_title"
+        ),
+        text: this.hass.localize(
+          "ui.panel.config.cloud.account.google.not_configured_text"
```

Comment:
This error message will only match the 404 status code, other errors should have a different message.